public/src/__tests__/TestHelpers.ts

```diff
@@ -72,12 +73,35 @@ export const setupDrivers = (path?: string): ThenableWebDriver[] => {
  * @param mockedEffects the mocked effects to initialize the mock with
  * NOTE: Directly setting derived values in the state is not supported.
 */
-export const initializeOvermind = (state: Partial<State & SubType<{ review: Partial<ReviewState>; }, object>>, mockedEffects?: Partial<typeof config.effects["client"]>
-) => {
+export const initializeOvermind = (state: Partial<State & SubType<{ review: Partial<ReviewState>; }, object>>, mockedApi?: ApiClient) => {
     const overmind = createOvermindMock(config, {
-        client: mockedEffects,
+        api: mockedApi
     }, initialState => {
         Object.assign(initialState, state)
-    })
+    });
+    Object.assign(overmind.effects.api, mockedApi)
     return overmind
 }
+
+/** UnaryApiClient is a type that represents the ApiClient without streaming methods. */
+interface UnaryApiClient {
+    client: Omit<ApiClient["client"], "submissionStream">;
+}
+
+/** Methods is a type that represents the methods of the UnaryApiClient */
+type Methods = UnaryApiClient["client"];
+
+/** mock is a helper function that takes a method and a mocked function to run in place of the method.
+ *  It returns a function that can be used to replace the method in the ApiClient.
+ * @param _method the method to mock
+ * @param mockFn the function to run in place of the method
+ * @example: mock("getSubmission", async (req) => { return { error: null, message: new Submission() } })
+*/
+export function mock<T extends keyof Methods>(
+    _method: T,
```

Comment:
Why use `_method` here and in a few other places?

Reply:
`method` is not used in the function, but exists to provide better auto completion. Underscoring the parameter tells the compiler that not using the parameter is intentional.

In our `tsconfig` we've enabled `"noUnusedParameters": true,`. With this enabled we get compile errors if there are any unused parameters in our code. ESLint also reports any unused parameters / variables. Using underscores will bypass these restrictions.

Omitting `method: T` altogether would work as long as we provide the method as a type argument:

```ts
mock<"getSubmission">(async () => { ... }) // mockFn is typed correctly
```

However, when adding the method as a type argument you would not get auto completion / the list of all available methods.